libtransmission/span.h

```diff
@@ -0,0 +1,56 @@
+/*
+* This file Copyright (C) 2008-2014 Mnemosyne LLC
+*
+* It may be used under the GNU GPL versions 2 or 3
+* or any future license endorsed by Mnemosyne LLC.
+*
+*/
+
+#pragma once
+
+#ifndef __TRANSMISSION__
+#error only libtransmission should #include this header.
+#endif
+
+#include <array>
+#include <vector>
+
+#include "transmission.h"
+
+/// @brief Readonly non-owning view into a provided memory block with start pointer and size.
+/// In C++20 this appears in standard library as std::span and remotely similar usage.
+template <typename T>
+class Span
+{
+public:
+    Span(const T* ptr, size_t size)
```

Comment:
```suggestion
    Span(T const* ptr, size_t size)
```

For consistency with the rest of the codebase